kratos/solving_strategies/strategies/residualbased_linear_strategy.h

```diff
@@ -476,6 +476,11 @@ class ResidualBasedLinearStrategy
 
         return mA;
     }
+    
```

Comment:
this change touches one of the important classes. As such it must be discussed on its own (I for one, do not love the name GetDirectSystemMatrix) 

The correct way would have been to make a PR only with those changes. 

for the moment let's leave it here, but i'll block the PR until we agree this is correct.
@pooyan-dadvand @jcotela comments on this? also @AndreasWinterstein, i think you also needed this capability some time back...

I have 2:
1 - I would call this GetSystemMatrix
2 - I am not clear of what happens when we are doing 
     
     A = *mpA. I think (but i may be wrong) that this triggers a copy and not a reference.

if this is so i would have implemented it as

    
     const TSystemMatrixType& GetSystemMatrix()
     {
         return *mpA;

Reply:
Teh way you suggest is the way is already implemented, but it was giving me problems to compile in python because the type of matrix was condensed and I try with the condensed matrix, but that gave me errors. So that why i created this method

Reply:
Please remove the new method. We need instead to fix the old method if it is not working

Reply:
OK, I need help with the boost::python then

Reply:
tomorrow when you are here we take a look at this. prepara a small test to verify how the existing implementation fails

Reply:
The current test can do it, it just one line